### PR TITLE
feat(server-routes): group routes by type

### PR DESCRIPTION
**Pull request by arashsheyda**
showing server-rotues from other modules in `external` group
related to #252 

<details>
  <summary>With routes from other modules</summary>
  
image

</details>

<details>
  <summary>without</summary>
  
image

</details>

### CODE REVIEW

File: packages/devtools/src/server-rpc/server-routes.ts [hunk 1]
```diff
@@ -1,128 +1,61 @@
...
+    cache = (() => {
       if (!nitro)
         return []
-      const files = await Promise.all([
-        scanServerDir(nitro, 'api', scanFileHandler('api')),
-        scanServerDir(nitro, 'routes', scanFileHandler('route')),
-      ]).then(r => r.flat())
-      return files
+      return [
+        ...nitro.scannedHandlers.map(item => ({
```
Comment (arashsheyda) [inline packages/devtools/src/server-rpc/server-routes.ts:36]: not sure if this was the best way to do it, sorry

**Comment by antfu** [comment]
Can you help to set up a new playground for routes with an example of external routes under `/playgrounds`? It looks good but I am not sure I understand the concept of "external" routes, and if they are worth to put in a separate group. Thanks

**Comment by arashsheyda** [comment]
@antfu yes it's taking a lot of space 😄 what about something like this? Maybe we could add the ability of seperation by group to user, so they can drag and create custom groups for themselves?
image

**Comment by arashsheyda** [comment]
@antfu here is a demo with latest push
the code may want some refactoring, sorry and thank you

https://github.com/nuxt/devtools/assets/38922203/6e088735-754e-4b23-8e1c-6c8f6ae5c4a7

**Comment by antfu** [comment]
No need to sorry at all! Thanks a lot for your efforts on pushing many features of DevTools!

It's really impressive you implemented the grouping and dragging system! But I am not very sure if it can be very useful as it requires manually managing them (and also the configure is save locally that are not shared across the team).

I thought we could the grouping automatically based on the directories (like a tree view) so they can manage the groups by simply managing the file structure. Users may toggle between plain list or tree views as well.

**Comment by arashsheyda** [comment]
@antfu yes that make sense thanks :D, can you do the grouping automatically based on the directories whenever you have the time? :)

**Comment by arashsheyda** [comment]
here is a demo with grouping automatically based on the directories, thank you

https://github.com/nuxt/devtools/assets/38922203/c66ae56f-b1bb-4ff7-8e9b-ca55f692802a

**Comment by arashsheyda** [comment]
also is it a good idea to have color for collections?

image

**Comment by arashsheyda** [comment]
@antfu What do you think😁

**Comment by antfu** [comment]
I love the UI and the grouping you made! But I am a bit hesitant about the breaking changes made to the UI options. I extracted nitro part #286 and merged it first so we are not blocked by this.

About the grouping and dragging, do you find a use case that you will need to manually organize the groups? I think we could auto-infer based on the routes structure. So user only needs to organize the routes by the file structure, which becomes the single source of trust. We could put the runtime in a separate group.

As for coloring, for DevTools I am thinking of not making it too colorful unless it has some meaning. But maybe changing the icon based on the type of API might be interesting.

**Comment by arashsheyda** [comment]
@antfu oh thank you, no I removed the dragging and used auto-grouping based on the routes name

**Comment by antfu** [comment]
Is it possible to keep the UI options as-is? If we absolutely need that, we could discuss it in another PR.

**Comment by arashsheyda** [comment]
@antfu awww sorry I misunderstood, I got it now. Sure, thank you

**Pull request merged**